src/netspresso_trainer/pipelines/base.py

```diff
@@ -130,7 +130,7 @@ def test_step(self, batch):
         raise NotImplementedError
 
     @abstractmethod
-    def get_metric_with_all_outputs(self, outputs):
+    def get_metric_with_all_outputs(self, outputs, phase):
```

Comment:
Shall we type-hint with `Literal` if there is no assert check for `phase` string?

Reply:
Good. I added type-hint with `Literal` on `phase`.